### PR TITLE
Added NUsight Overview content

### PR DESCRIPTION
**Pull request by vuvuzella**
Add contents to System-Nusight Overview which describes the basic features of NUsight

Update: preview is now in commit 6e2d7f8de39aa98e286f76b8670c2987a3476afe
[preview](https://deploy-preview-51--nubook.netlify.app/system/tools/nusight)

### CODE REVIEW

File: src/book/02-system/02-nusight/01-overview.mdx [hunk 1]
```diff
@@ -5,3 +5,101 @@ title: Overview
 description: Introduction to NUsight and how to setup your environment to run it.
 slug: /system/nusight/overview
 ---
+
+## NUsight Visualization System
+
+### Overview
+
+The NUsight visualization system is a web-based visualizer created in-house for the NUbots team. It provides real time display of information of messages passed inside the NUclear robotics framework. The data that can be visualized in NUsight may come from the real hardware or simulated hardware.
```
Comment (ysims) [inline src/book/02-system/02-nusight/01-overview.mdx:13]:
```suggestion
The NUsight visualization system is a web-based visualizer created in-house for the NUbots team. It provides real time display of information of messages passed inside the NUClear robotics framework. The data that can be visualized in NUsight may come from the real hardware or simulated hardware.
```

File: src/book/02-system/02-nusight/01-overview.mdx [hunk 2]
```diff
@@ -5,3 +5,101 @@ title: Overview
...
+
+Dasboard displays a bird's eye view of the playing field with the robots loacalized in the field. This section also displays the overall state of all robots.
```
Comment (ysims) [inline src/book/02-system/02-nusight/01-overview.mdx:15]:
```suggestion
Dashboard displays a bird's eye view of the playing field with the robots localised in the field. This section also displays the overall state of all robots.
```

File: src/book/02-system/02-nusight/01-overview.mdx [hunk 3]
```diff
@@ -5,3 +5,101 @@ title: Overview
...
+
+![Dashboard](./images/dashboard-main.png)
+
+Flip Orientation flips the playing field to display the mirror image of the previous view of the playing field.
+
+#### Playing Field
+
+![Dashboard](./images/nusight-dashboard.png)
+
+The playing field displays all the robots that are currently connected to NUsight. It. The yellow rectangle indicates our own goal. The blue one is the opponent's goal.
```
Comment (ysims) [inline src/book/02-system/02-nusight/01-overview.mdx:37]:
```suggestion
The playing field displays all the robots that are currently connected to NUsight. The yellow rectangle indicates our own goal. The blue one is the opponent's goal.
```

File: src/book/02-system/02-nusight/01-overview.mdx [hunk 4]
```diff
@@ -5,3 +5,101 @@ title: Overview
...
+
+Each robot displays the number assigned to it by NUsight. The wedge on the representation of the robot indicates the direction of the robot's orientation. The black arcs coming out of each robot indicates the robot's walk path
+
+The orange dot in the playing field indicates where the robot thinks the ball is located. The green arrow coming out of the orange ball indicates how the robot will kick the ball. Usually it should point towards the opponent's goal.
```
Comment (ysims) [inline src/book/02-system/02-nusight/01-overview.mdx:41]: Maybe make these a list or a table

File: src/book/02-system/02-nusight/01-overview.mdx [hunk 5]
```diff
@@ -5,3 +5,101 @@ title: Overview
...
+
+| Keyboard | Action                                                         |
+| :------: | -------------------------------------------------------------- |
+|    w     | Move forward                                                   |
+|    s     | Move backward                                                  |
+|    a     | Strafe left                                                    |
+|    d     | Strafe right                                                   |
+|  Space   | Switch between Free Camera, First Person or Thrid Persion view |
+
+|       Mouse       | Action                                          |
```
Comment (ysims) [inline src/book/02-system/02-nusight/01-overview.mdx:73]: maybe concatenate these tables

File: src/book/02-system/02-nusight/01-overview.mdx [hunk 6]
```diff
@@ -5,3 +5,101 @@ title: Overview
...
+- compass
+- Horizon
+- Visual Mesh
+- Green Horizon
+- Balls
+- Goals
+
+### Visual Mesh
+
+This section displays the data from the robot's camera and provides a mesh overlay on the image using the Visual Mesh object detection algorithm
```
Comment (ysims) [inline src/book/02-system/02-nusight/01-overview.mdx:105]: We should get an image for this, maybe @TrentHouliston has one?

Comment (vuvuzella) [reply src/book/02-system/02-nusight/01-overview.mdx:105]: Can I use the images from, the one with mesh on it https://github.com/Fastcode/VisualMesh/tree/4a64fe3b0b21dacf09d644dee72e9c6d6408a1d3? or should it be the visual mesh inside nusight>

Comment (ysims) [reply src/book/02-system/02-nusight/01-overview.mdx:107]: That image won't work, what you see in NUsight is a little different. Will try to get an image.

File: src/book/02-system/03-tools/01-nusight.mdx [hunk 7]
```diff
@@ -3,5 +3,102 @@ section: System
 chapter: Tools
 title: NUsight
 description: NUsight and how to setup your environment to run it.
-slug: /system/tools/overview
+slug: /system/tools/nusight
 ---
+
+The NUsight visualisation system is a web-based visualiser created in-house for the NUbots team. It provides real-time display of information of messages passed inside the NUClear robotics framework. The data that can be visualised in NUsight may come from real hardware or simulated hardware. NUsight contains multiple tabs with different debugging capabilities.
```
Comment (JosephusPaye) [inline src/book/02-system/03-tools/01-nusight.mdx:9]:
```suggestion
The NUsight visualisation tool is a web-based visualiser created in-house for the NUbots team, to aid in building and debugging our systems. It provides real-time display of information from other systems using [NUClear networking](/system/foundations/nuclear). The data visualised in NUsight may come from real hardware or simulated hardware. The NUsight interface contains multiple tools in various "tabs" with different debugging capabilities.
```

File: src/book/02-system/03-tools/01-nusight.mdx [hunk 8]
```diff
@@ -3,5 +3,102 @@ section: System
...
+
+The 'Select Robots' button in the top right corner enables the display of specific real or simulated robots. This affects all the tabs in NUsight.
+
+## Dashboard
+
+![NUsight with Dashboard selected. Top half of the screen is a birds-eye view of the field. Contains symbols representing robots, balls, walk paths and kick paths. The bottom half has text specifying the state of each robot.](./images/dashboard-main.png)
+
+The Dashboard gives an overview of the state of the robots on the field.
+
+The 'Flip Orientation' button in the top left hand corder flips the playing field to display the mirror image of the previous view of the playing field.
```
Comment (JosephusPaye) [inline src/book/02-system/03-tools/01-nusight.mdx:27]:
```suggestion
The 'Flip Orientation' button in the top left hand corner flips the playing field to display the mirror image of the previous view of the playing field.
```

File: src/book/02-system/03-tools/01-nusight.mdx [hunk 9]
```diff
@@ -3,5 +3,102 @@ section: System
...
+- Camera, Ball and Goal Icons
+  - Flash Orange - a new packet is received
+  - Red Outline box - Indicates that a long time has passed since the last packet of data was received.
+  - Greyed Out - Inactive
+
+## Localisation
+
+![Localisation tab. Shows a 3D scene with a soccer field and NUgus robots.](./images/localisation-main.png)
+
+The localisation view shows a 3D model of any connected robots with their position and orientation on the field.
```
Comment (JosephusPaye) [inline src/book/02-system/03-tools/01-nusight.mdx:62]:
```suggestion
The localisation view shows a 3D model of any connected robots with their position and orientation on the field. It is interactive and can be used with the following commands:
```

File: src/book/02-system/03-tools/01-nusight.mdx [hunk 10]
```diff
@@ -3,5 +3,102 @@ section: System
...
+## Chart
+
+![Chart tab. Shows a coordinate plane with multiple graphs plotted, each with different colours.](./images/chart-main.png)
+
+The chart view shows a coordinate plane with a variable graphed against time. Multiple graphs can be overlayed on top of each other.
+
+| Feature               | Description                                                                                                                       |
+| --------------------- | --------------------------------------------------------------------------------------------------------------------------------- |
+| Line Chart/2D Scatter | Switch between a line chart or a scatter plot to represent the data on the chart. Currently the scatter plot is unsupported.      |
+| Right Panel           | Select which variable from which robot is to be plotted on the chart.                                                             |
```
Comment (JosephusPaye) [inline src/book/02-system/03-tools/01-nusight.mdx:82]:
```suggestion
| Right Panel           | Select specific datapoints from specific robots to be plotted on the chart. Also allows for changing the color of a line on the chart. |
```

**Pull request merged**